frontend/src/v5/ui/routes/viewer/tickets/ticketsList/ticketsList.component.tsx

```diff
@@ -69,22 +69,20 @@ export const TicketsList = ({ tickets }: TicketsListProps) => {
 		return (isCompletedIssueProperty || isCompletedTreatmentStatus) === showingCompleted;
 	};
 
-	useEffect(() => {
-		const filtered = tickets.filter((ticket) => {
-			const matchesTemplateFilters = !selectedTemplates.size || selectedTemplates.has(ticket.type);
-			return (matchesTemplateFilters && matchesCompletedState(ticket));
-		});
-		setFilteredTickets(filtered);
-	}, [tickets, selectedTemplates, showingCompleted]);
+	const applyTemplatesFilter = (items) => items.filter((ticket) => !selectedTemplates.size || selectedTemplates.has(ticket.type));
```

Comment:
Can we rename this to something like "filterByTemplate"?

Also, you can check if there are no selected templates just once for all the tickets:

```ts
	const applyTemplatesFilter = (items) => {
		if (!selectedTemplates.size) return items;
		return items.filter((ticket) => selectedTemplates.has(ticket.type));
	};
```